lib/galaxy/selenium/navigates_galaxy.py

```diff
@@ -480,16 +526,20 @@ def _latest_history_item(self) -> Optional[dict[str, Any]]:
             return None
 
     def wait_for_history(self, assert_ok=True):
-        def history_becomes_terminal(driver):
+        def history_becomes_terminal(driver=None):
             current_history_id = self.current_history_id()
             state = self.api_get(f"histories/{current_history_id}")["state"]
             if state not in ["running", "queued", "new", "ready"]:
                 return state
             else:
                 return None
 
-        timeout = self.timeout_for(wait_type=WAIT_TYPES.JOB_COMPLETION)
-        final_state = self.wait(timeout=timeout).until(history_becomes_terminal)
+        timeout = self.wait_length(wait_type=WAIT_TYPES.JOB_COMPLETION)
+        if self.backend_type == "playwright":
+            final_state = wait_on(history_becomes_terminal, "history to become terminal", timeout)
+        else:
```

Comment:
Can we not express this the same way in one of the parent classes ?

Reply:
We can in fact - there is a TODO that remains about this in the playwright limitations but I should done this there instead of hacking on it a layer up - good catch.